artemis/module_base.py

```diff
@@ -204,9 +213,28 @@ def _take_and_lock_tasks(self, num_tasks: int) -> Tuple[List[Task], List[Optiona
         try:
             tasks = []
             locks: List[Optional[ResourceLock]] = []
-            for queue in self.backend.get_queue_names(self.identity):
-                self.log.debug(f"[taking tasks] Taking tasks from queue {queue}")
-                for i, item in enumerate(self.backend.redis.lrange(queue, 0, -1)):
+
+            if self.queue_location_timestamp < time.time() - self.queue_location_max_age_seconds:
+                self.queue_id = 0
+                self.queue_position = 0
+                self.queue_location_timestamp = time.time()
```

Comment:
I guess, best would be if `time.time()` was assigned to variable before if statement, then it will match timestamp used in conditional

Reply:
I don't see any reason to do that, I want to update the timestamp simply to current time, no problem if the two time..time() calls diverge